logic/atrium-logic/src/commonMain/kotlin/ch/tutteli/atrium/logic/creating/charsequence/contains/searchers/impl/RegexSearcher.kt

```diff
@@ -11,10 +11,14 @@ import ch.tutteli.atrium.logic.creating.charsequence.contains.searchbehaviours.N
 class RegexSearcher : Searcher<NoOpSearchBehaviour, Regex> {
     override fun search(searchIn: CharSequence, searchFor: Regex): Int {
         var counter = 0
-        var matchResult = searchFor.find(searchIn)
+        val searchInString = searchIn.toString()
```

Comment:
Just wondering: does `searchIn.findAll(searchFor): Sequence<MatchResult>` work?
`MatchResult.next(): MatchResult?` work?

It looks a bit weird to hand-craft it

Reply:
findAll would be a disjoint search, i.e. search `aa` in `aaaa` would find 2 occurrences. we use a non-disjoint search i.e. it finds it 3 times. Don't ask me why I went with a non-disjoint search tough. Looking at it now it seems a bit weird but back in 2017 I had a use case where it was relevant and it seemed natural to implement it that way. I guess it was an edge case as I cannot remember it. Most times it probably doesn't matter, so... I would keep it that way

Reply:
MatchResult.next() and friends go for disjoint matches which is what I would have expected if I used `toContain o exactly 3 regex`.

If you implement overlapping occurrences, then the hand-rolled impl is ok. I was just curios, there were not comments, so I asked.

Reply:
> disjoint matches which is what I would have expected if I used toContain o exactly 3 regex.

I get that as I had the same assumption some time ago. It's documented in the KDoc (also for `toContain o exactly 3 regex`) but it's not intuitive. I guess I will change the behaviour with 2.0.0